components/Elevation/tests/unit/MaterialElevationColorTests.m

```diff
@@ -0,0 +1,256 @@
+
+  // When
+  UIColor *resolvedRGBColor = [self.rgbColor mdc_resolvedColorWithElevation:elevation];
+  UIColor *resolvedGreyScaleColor = [self.greyScaleColor mdc_resolvedColorWithElevation:elevation];
+  UIColor *resolvedP3DisplayColor = [self.p3DisplayColor mdc_resolvedColorWithElevation:elevation];
+
+  // Then
+  UIColor *expectedRGBColor = [UIColor colorWithRed:(CGFloat)0.91964261807423053
+                                              green:(CGFloat)0.83928523614846129
+                                               blue:(CGFloat)0.67857047229692247
+                                              alpha:(CGFloat)0.65116211491037057];
+  ;
+  UIColor *expectedGreyScaleColor = [UIColor colorWithRed:(CGFloat)0.91964261807423053
+                                                    green:(CGFloat)0.91964261807423053
+                                                     blue:(CGFloat)0.91964261807423053
+                                                    alpha:(CGFloat)0.65116211491037057];
+  ;
+  UIColor *expectedP3Display = [UIColor colorWithRed:(CGFloat)0.86849367970437186
+                                               green:(CGFloat)0.77713910118968843
+                                                blue:(CGFloat)0.61272176809458923
+                                               alpha:(CGFloat)0.47674317236555602];
+  XCTAssertTrue([self compareColorsWithFloatPrecisionFirstColor:resolvedRGBColor
+                                                    secondColor:expectedRGBColor],
+                @"(%@) is not equal to (%@)", resolvedRGBColor, expectedRGBColor);
+  XCTAssertTrue([self compareColorsWithFloatPrecisionFirstColor:resolvedGreyScaleColor
+                                                    secondColor:expectedGreyScaleColor],
+                @"(%@) is not equal to (%@)", resolvedGreyScaleColor, expectedGreyScaleColor);
+  XCTAssertTrue([self compareColorsWithFloatPrecisionFirstColor:resolvedP3DisplayColor
+                                                    secondColor:expectedP3Display],
+                @"(%@) is not equal to (%@)", resolvedP3DisplayColor, expectedP3Display);
+}
+
+- (void)testResolvedColorWithHighElevation {
+  // Given
+  CGFloat elevation = (CGFloat)10000;
+
+  // When
+  UIColor *resolvedRGBColor = [self.rgbColor mdc_resolvedColorWithElevation:elevation];
+  UIColor *resolvedGreyScaleColor = [self.greyScaleColor mdc_resolvedColorWithElevation:elevation];
+  UIColor *resolvedP3DisplayColor = [self.p3DisplayColor mdc_resolvedColorWithElevation:elevation];
+
+  // Then
+  UIColor *expectedRGBColor = [UIColor colorWithRed:(CGFloat)0.95614843571155961
+                                              green:(CGFloat)0.91229687142311943
+                                               blue:(CGFloat)0.82459374284623876
+                                              alpha:(CGFloat)0.77378792660557727];
+  ;
+  UIColor *expectedGreyScaleColor = [UIColor colorWithRed:(CGFloat)0.95614843571155961
```

Comment:
It is caused by the blending is all done in RGB. We can resolve/improve it after addressing the blending method we use in library.

Reply:
Ok. Please document the change in color space as a TODO and file an Issue. Since we're blending with `UIColor.whiteColor`, I think grayscale should be supported eventually.

Reply:
Addressed in f3b10263cdb82f6e33f7001924ba22743df120a9